#
# This patch fixes freedesktop.org bug #6820, which makes Xephyr randomly die
#
# https://bugs.freedesktop.org/show_bug.cgi?id=6820
#
# Written by Alan Coopersmith
#
diff --git a/src/XlibInt.c b/src/XlibInt.c
index 3a1f96f..56bacba 100644
--- a/src/XlibInt.c
+++ b/src/XlibInt.c
@@ -135,11 +135,7 @@ #ifdef __UNIXOS2__
 #define ECHECK(err) (errno == err)
 #define ESET(val)
 #else
-#ifdef ISC
-#define ECHECK(err) ((errno == err) || ETEST())
-#else
 #define ECHECK(err) (errno == err)
-#endif
 #define ESET(val) errno = val
 #endif
 #endif
@@ -297,7 +293,7 @@ #else
 	    nfound = Select (dpy->fd + 1, &r_mask, &w_mask, NULL, NULL);
 #endif
 	    InternalLockDisplay(dpy, cv != NULL);
-	    if (nfound < 0 && !ECHECK(EINTR))
+	    if (nfound < 0 && !(ECHECK(EINTR) || ETEST()))
 		_XIOError(dpy);
 	} while (nfound <= 0);
 
@@ -509,7 +505,7 @@ #else
 	result = Select(highest_fd + 1, &r_mask, NULL, NULL, NULL);
 #endif
 	InternalLockDisplay(dpy, dpy->flags & XlibDisplayReply);
-	if (result == -1 && !ECHECK(EINTR)) _XIOError(dpy);
+	if (result == -1 && !(ECHECK(EINTR) || ETEST())) _XIOError(dpy);
 	if (result <= 0)
 	    continue;
 #ifdef USE_POLL
@@ -826,7 +822,7 @@ #endif
 		    if (!pend)
 			pend = SIZEOF(xReply);
 		}
-		else if (result < 0 && !ECHECK(EINTR))
+		else if (result < 0 && !(ECHECK(EINTR) || ETEST()))
 		    _XIOError(dpy);
 	    }
 	}

